mimic/model/identity_objects.py

```diff
@@ -62,17 +80,164 @@ def not_found(message, request):
+        are used for creating an instance of the :obj:`ExternalApiStore`
+        that will enable the listing to be put into the Service Catalog;
+        the `id` and `region` fields are used by the :obj:`ExternalApiStore`
+        for managing the templates.
+
+    [0] http://developer.openstack.org/api-ref-identity-v2-ext.html
+    """
+
+    """
+    The following lists of tuples are for mapping the template defined by
+    the OpenStack documentation to the internal strucutre for several uses.
+    The tuples have the following meaning:
+
+        Index   Meaning
+        -----   ------------------------------------------------
+        0       JSON Key in the OpenStack Identity Template Spec
+        1       Attribute Name of the :obj:`EndpointTemplateStore`
+        2       default value (only for optional_mapping)
+    """
+    required_mapping = [
+        ('id', 'id_key'),
+        ('region', 'region_key'),
+        ('type', 'type_key'),
+        ('name', 'name_key'),
+    ]
+
+    optional_mapping = [
+        ('enabled', 'enabled_key', False),
+        ('publicURL', 'public_url', u""),
+        ('internalURL', 'internal_url', u""),
+        ('adminURL', 'admin_url', u""),
+        ('RAX-AUTH:tenantAlias', 'tenant_alias', "%tenant_id%"),
+        ('versionId', 'version_id', u""),
+        ('versionInfo', 'version_info', u""),
+        ('versionList', 'version_list', u"")
+    ]
+
+    tenant_mapping = [
+        ('id', 'id_key'),
+        ('region', 'region_key'),
+        ('type', 'type_key'),
+        ('publicURL', 'public_url'),
+        ('internalURL', 'internal_url'),
+        ('adminURL', 'admin_url'),
+    ]
+
+    def __init__(self, template_dict=None):
+        """
+        Create an :obj:`EndpointTemplateStore`
+
+        :param dict template_dict (optional): optional dictionary to load the
+            Endpoint Template data from. The dictionary is simply passed to the
+            deserializer.
+        """
+        self._template_data = template_dict
+        self.id_key = None
+        self.region_key = None
+        self.type_key = None
+        self.name_key = None
+        self.enabled_key = None
+        self.public_url = None
+        self.internal_url = None
+        self.admin_url = None
+        self.tenant_alias = None
+        self.version_id = None
+        self.version_info = None
+        self.version_list = None
+
+        if self._template_data is not None:
+            self.deserialize(self._template_data)
+
+    def get_url(self, url, tenant_id):
+        """
+        Apply the tenant_id replacement to the URL.
+
+        :param text_type url: the URL to do the replacement on.
+        :param text_type tenant_id: the tenant-id to insert into the URL.
+        """
+        value_to_replace = self.tenant_alias
+        if value_to_replace is None:
+            value_to_replace = '%tenant_id%'
+
+        return url.replace(value_to_replace, tenant_id)
+
+    def serialize(self, tenant_id=None):
```

Comment:
If this were an `attr.s`-ified class, it could use [`asdict`](https://attrs.readthedocs.io/en/stable/api.html#attr.asdict) to do serialization and skip most of this code, I believe.